helix-view/src/clipboard/config.rs

```diff
@@ -0,0 +1,248 @@
+    test_commands: Vec<Vec<String>>,
+}
+
+macro_rules! command_provider_spec {
+    (
+        copy: $($get:literal),+ ;
+        paste: $($set:literal),+ ;
+        $(
+            primary_copy: $($pget:literal),+ ;
+            primary_paste: $($pset:literal),+ ;
+        )?
+        $( env: $($env:literal),+ ; )?
+        $( test: $($test:literal),+ ; )*
+    ) => {
+        #[allow(clippy::needless_update)]
+        ClipboardProviderSpec {
+            copy: vec![$($get.to_owned()),+],
+            paste: vec![$($set.to_owned()),+],
+            $(
+                primary_copy: Some(vec![$($pget.to_owned()),+]),
+                primary_paste: Some(vec![$($pset.to_owned()),+]),
+            )?
+            $( env_vars: vec![$($env.to_owned()),+], )?
+            test_commands: vec![$(vec![$($test.to_owned()),+]),*],
+            ..Default::default()
+        }
+    };
+}
+
+macro_rules! clipboard_config {
+    ($(
+        $(#[$attr:meta])*
+        $name:literal {
+            $($body:tt)*
+        }
+    )*) => {
+        ClipboardConfig {
+            providers: HashMap::from([$(
+                $(#[$attr])*
+                ($name.to_owned(), command_provider_spec! { $($body)* })
+            ),*]),
+            order: vec![$(
+                $(#[$attr])*
+                $name.to_owned()
+            ),*],
+        }
+    };
+}
+
+impl Default for ClipboardConfig {
+    fn default() -> Self {
+        let mut cfg = clipboard_config! {
+            #[cfg(unix)]
+            "wl-clipboard" {
+                copy: "wl-copy", "--type", "text/plain";
+                paste: "wl-paste", "--no-newline";
+                primary_copy: "wl-copy", "-p", "--type", "text/plain";
+                primary_paste: "wl-paste", "-p", "--no-newline";
+                env: "WAYLAND_DISPLAY";
+            }
+
+            #[cfg(unix)]
+            "xclip" {
+                copy: "xclip", "-i", "-selection", "clipboard";
+                paste: "xclip", "-o", "-selection", "clipboard";
+                primary_copy: "xclip", "-i";
+                primary_paste: "xclip", "-o";
+                env: "DISPLAY";
+            }
+
+            #[cfg(unix)]
+            "xsel" {
+                copy: "xsel", "-o", "-b";
+                paste: "xsel", "-i", "-b";
+                primary_copy: "xsel", "-o";
+                primary_paste: "xsel", "-i";
+                env: "DISPLAY";
+                test: "xsel", "-o", "-b";
+            }
+
+            "lemonade" {
+                copy: "lemonade", "copy";
+                paste: "lemonade", "paste";
+            }
+
+            #[cfg(unix)]
+            "doit" {
+                copy: "doitclient", "wclip";
+                paste: "doitclient", "wclip", "-r";
+            }
+
+            #[cfg(any(windows, target_os = "linux"))] // this is a godsend on WSL
+            "win32yank" {
+                copy: "win32yank.exe", "-i";
+                paste: "win32yank.exe", "-o";
+            }
```

Comment:
Looking at what the code was before, you want to always use the `WindowsProvider` on `windows`, and `win32yank` on `linux` which will cover WSL.

Reply:
Sure, my particular use case doesn't need anything other than `clipboard-win` on `cfg(windows)`, but I believe the `lemonade` client also works on Windows.